src/autorelease.rs

```diff
@@ -0,0 +1,38 @@
+use std::os::raw::c_void;
+use runtime::{objc_autoreleasePoolPush, objc_autoreleasePoolPop};
+
+struct AutoReleaseHelper {
+    context: *mut c_void
+}
+
+impl AutoReleaseHelper {
+    fn new() -> AutoReleaseHelper {
+        unsafe { AutoReleaseHelper{ context: objc_autoreleasePoolPush()} }
+    }
+}
+
+impl Drop for AutoReleaseHelper {
+    fn drop(&mut self) {
+        unsafe { objc_autoreleasePoolPop(self.context) }
+    }
+}
+
+/**
+Execute `f` in the context of a new autorelease pool. The pool is drained
+after the execution of `f` completes. This corresponds to @autoreleasepool blocks
+in Objective-c and Swift.
+*/
+pub fn autoreleasepool<F: FnOnce()>(f: F) {
```

Comment:
perhaps, we should just make it a method of `AutoReleaseHelper`?

Reply:
Nope. That would be unsafe because we can't guarantee that drop() will be called.